eZ/Publish/Core/MVC/Symfony/Controller/Content/DownloadController.php

```diff
@@ -42,12 +42,19 @@ public function __construct(ContentService $contentService, IOService $ioService
      * @param \Symfony\Component\HttpFoundation\Request $request
      *
      * @return \eZ\Bundle\EzPublishIOBundle\BinaryStreamResponse
-     * @return \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
      */
     public function downloadBinaryFileAction($contentId, $fieldIdentifier, $filename, Request $request)
     {
         if ($request->query->has('version')) {
-            $content = $this->contentService->loadContent($contentId, null, $request->query->get('version'));
+            $version = (int) $request->query->get('version');
+            if ($version <= 0) {
+                throw new NotFoundException('File', $filename);
+            }
+            $content = $this->contentService->loadContent($contentId, null, $version);
```

Comment:
```suggestion
            $version = $request->query->getInt('version');
            if ($version <= 0) {
                throw new NotFoundException('File', $filename);
            }
            
            $content = $this->contentService->loadContent($contentId, null, $version);
```

Reply:
or potentially \`\\Symfony\\Component\\HttpFoundation\\Exception\\BadRequestException\`

Reply:
Thanks. Clearing the question of whether it potentially can be accepted for merge, before applying this.